aip/client-libraries/4236.md

```diff
@@ -0,0 +1,36 @@
+---
+id: 4236
+state: approved
+created: 2024-05-16
+---
+
+# Version-aware clients
```

Comment:
I've made a bunch of comments re:guidance appearing in this preamble - perhaps that means the preamble needs to be cut down/generalized/removed/converted into guidance. We don't want to repeat ourselves, we don't want to spread guidance out into various places, and we want to be succinct. Adding a `Rationale` section go/aip/8#rationale is probably where you want to be explaining the "why"

Reply:
Thanks for reviewing @noahdietz ! Please let me know if this latest version reads better.

Reply:
I does! A few more tweaks thanks Tony :)

Reply:
Please could you take another look?